compiler/rustc_typeck/src/check/method/mod.rs

```diff
@@ -443,6 +489,123 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         Some(InferOk { obligations, value: callee })
     }
 
+    pub(super) fn lookup_op_method_in_trait(
+        &self,
+        span: Span,
+        m_name: Ident,
+        trait_def_id: DefId,
+        self_ty: Ty<'tcx>,
+        opt_input_type: Option<Ty<'tcx>>,
+        opt_input_expr: Option<&'tcx hir::Expr<'tcx>>,
+    ) -> Option<InferOk<'tcx, MethodCallee<'tcx>>> {
```

Comment:
This whole method is very similar to `lookup_method_in_trait`, right? Could we have a third method that has the common parts that gets called from `lookup_method_in_trait` and `lookup_op_method_in_trait`? Duplicated code that's supposed to do the same thing will be a source of bugs in the future as people make small changes to one method and not the other and they start to diverge accidentally.

Reply:
> Could we have a third method that has the common parts that gets called from lookup_method_in_trait and lookup_op_method_in_trait?

Yes. I am going to implement it!